tests/test_react_auth_defs.py

```diff
@@ -121,22 +122,11 @@ def _change_random_character_within_string_ntimes(value: str, n: int) -> str:
     return value
 
 
-def mock_foursight_env_name(env):
-    if env.startswith("env-a"):
-        return "env-a-foursight-name"
-    elif env.startswith("env-b"):
-        return "env-b-foursight-name"
-    elif env.startswith("env-c"):
-        return "env-c-foursight-name"
-    else:
-        return env
-
-def mock_short_env_name(env):
-    if env.startswith("env-a"):
-        return "env-a-short-name"
-    elif env.startswith("env-b"):
-        return "env-b-short-name"
-    elif env.startswith("env-c"):
-        return "env-c-short-name"
-    else:
-        return env
+def mock_foursight_env_name(env: str) -> Optional[str]:
+    known_env = [known_env for known_env in KNOWN_ENVS if env in known_env.values()]
+    return known_env[0].get("foursight_name") if len(known_env) == 1 else None
+
+
+def mock_short_env_name(env: str) -> Optional[str]:
+    known_env = [known_env for known_env in KNOWN_ENVS if env in known_env.values()]
+    return known_env[0].get("short_name") if len(known_env) == 1 else None
```

Comment:
Yeah, that will work. But what I was thinking requires no setup. (There is a subtle difference between yours and what I'm thinking of, because yours sometimes returns `None` sometimes. `full_env_name` never does that. `short_env_name` only does it (for historical reasons) if given `None` as an argument, but never if given a string. So it depends on what you expect of your mock. (My mock could be adjusted to make the short env mock more correct by adding a check for `None` that returns `None`. I wrote it the way I did below to emphasize the symmetry.)

My main and more pressing concern, though, is wondering why we're using KNOWN_HOSTS to get full and short names. This is just not how it should be done. It should be using env_utils operations so that the entire system has consistent interpretation of names. Doing it by an ad hoc means will lead to odd behaviors that are not consistent throughout the product.  This is not so much about the mocking, but about the actual production code. The way we mock this kind of behavior in env_utils is to bind the cached EnvUtils declarations to a local ecosystem description, and then we get consistent behavior of everything.

I think I've commented on this before. If we aren't going to fix it for this PR (and we could bumble along with an approximation like you have here for now, I don't want to hold up this PR over it), then please write a ticket and queue it for doing soon after.

```
from dcicutils.misc_utils import remove_suffix
def assure_suffix(suffix, s):  # could go in misc_utils
    return s if s.endswith(suffix) else s + suffix

SHORT_NAME_SUFFIX = "-short-name"

def mock_foursight_env_name(env: str):
    return remove_suffix(SHORT_NAME_SUFFIX, env)

def mock_foursight_short_name(env: str):
    return assure_suffix(SHORT_NAME_SUFFIX, env)
```

Reply:
Yeah, sorry, I'm not 100% sure I'm precisely getting your more general point. Created this ticket to dig into it: https://hms-dbmi.atlassian.net/browse/C4-953